topi/python/topi/nn/upsampling.py

```diff
@@ -1,25 +1,36 @@
 """TVM operator upsampling compute."""
 from __future__ import absolute_import
-import tvm
-from .. import util
+import topi
 
 
-def upsampling(data, scale, layout="NCHW"):
-    """Perform nearest neighbor upsampling on the data.
-       Bilinear upsampling is not supported.
+def upsampling(data, scale, layout="NCHW", align_corners=False, mode='NN', weights=None):
+    """Perform upsampling on the data.
+       Nearest neighbourhood and bilinear are supported.
 
     Parameters
     ----------
-    data : tvm.Tensor
-        4-D with shape [batch, channel, in_height, in_width]
+    inputs : tvm.Tensor
+        inputs is a 4-D tensor with shape
+        [batch, channel, in_height, in_width]
         or  [batch, in_height, in_width, channel]
 
-    scale: int
-        upsampling scaling factor
+    scale :
+        Scaling factor
 
-    layout: string
+    layout : string
         either "NCHW" or "NHWC"
 
+    align_corners : Boolean
+        To preserve the values at the corner pixels
+
+    mode : string
+        either "NN" or "BILINEAR"
+
+    weights : tvm.Tensor
```

Comment:
can you just call your bilinear_weights() function here, instead of requiring users to provide weights?

Reply:
Weights calculation require data shape which is a placeholder tensor here (no dimension specified).
Other reason to precompute weights is to avoid the same compute on runtime for (NxC) times.
More discussion @ https://discuss.tvm.ai/t/scaling-design-for-nn-nearest-neighbour-and-bilinear/187/22